virttest/utils_time.py

```diff
@@ -0,0 +1,259 @@
+            # Empty line
+            if not line:
+                continue
+            _code = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line)
+            if _code and _code.group(1):
+                _code_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2',
+                                _code.group(1))
+                timezone_set.append(_code_)
+                continue
+            # When UTC standard time, add timezone code '+0000'
+            elif _code:
+                timezone_set.append("+0000")
+                continue
+            _name = re.search(r'(\S+(?:\s\S+)*)', line)
+            if _name:
+                timezone_set.append(_name.group(0))
+            else:
+                logging.warn("Can not get timezone name correctly.")
+            timezone_sets.append(timezone_set)
+            timezone_set = []
+        return timezone_sets
+
+    def get_timezone_code(timezone_name):
+        for value in get_timezone_list():
+            if value[1] == timezone_name:
+                return value[0]
+        return None
+
+    def get_timezone_name(timezone_code):
+        for value in get_timezone_list():
+            if value[0] == timezone_code:
+                return value[1]
+        return None
+
+    error_context.context("Verify guest's timezone", logging.info)
+    timezone_cmd = 'tzutil /g'
+    host_timezone_code = get_host_timezone()['timezone_code']
+    timezone_name = session.cmd_output(timezone_cmd).strip('\n')
+    if get_timezone_code(timezone_name) != host_timezone_code:
+        return get_timezone_name(host_timezone_code)
+    return True
+
+
+@error_context.context_aware
+def sync_timezone_win(vm):
+    """
+    Verify and sync windows guest's timezone
+
+    :param vm: Virtual machine for vm
+    """
+    session = vm.wait_for_login()
+    set_timezone_cmd = 'tzutil /s "%s"'
+    verify_result = verify_timezone_win(session)
+
+    if verify_result is not True:
+        error_context.context("Sync guest's timezone.", logging.info)
+        session.cmd(set_timezone_cmd % verify_result)
+        vm_params = vm.params
+        error_context.context("Shutdown guest...", logging.info)
+        vm.destroy()
+        error_context.context("Boot guest...", logging.info)
+        vm.create(params=vm_params)
+        vm.verify_alive()
+        session = vm.wait_for_login()
+        if verify_timezone_win(session) is not True:
+            raise exceptions.TestError("Fail to sync guest's timezone.")
+    session.close()
+
+
+def execute(cmd, timeout=360, session=None):
+    """
+    Execute command in guest or host, if session is not None return
+    command output in guest else return command ouput in host
+
+    :param cmd: Shell commands
+    :param timeout: Timeout to execute command
+    :param session: ShellSession or None
+
+    :return: Command output string
+    """
+    if session:
+        ret = session.cmd_output(cmd, timeout=timeout)
+    else:
+        ret = decode_to_text(process.system_output(cmd))
```

Comment:
`cmd` from `sync_time_with_ntp()`requires `shell=True` so it is better to use `process.getoutput()` that takes care of it.